rootfs/etc/nginx/lua/configuration.lua

```diff
@@ -110,6 +110,36 @@ local function handle_general()
   ngx.status = ngx.HTTP_CREATED
 end
 
+local function handle_certs()
+  if ngx.var.request_method ~= "GET" then
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    ngx.print("Only GET requests are allowed!")
+    return
+  end
+
+  local query = ngx.req.get_uri_args()
+  if not query["hostname"] then
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    ngx.print("Hostname must be specified.")
+    return
+  end
+
+  local key = _M.get_pem_cert_key(query["hostname"])
+  if key and key ~= "" then
```

Comment:
> key ~= ""

I don't think this should be "not found" - it's found, just empty.

Reply:
So if the queried host isn't configured at all, this is nil. If the host exists in the ingress definition at all but doesn't have a certificate, this is the empty string. I feel like we should return 404 for both of these cases.

Reply:
Based on some in-person discussion, I've made this change.

Reply:
tldr; returning an empty content in that case reveals more information than returning 404.